### DISCUSSION TITLE
Information on character (fontname, adv)

### DISCUSSION THREAD

**Discussion by EmanueleGusso** [jsvine/pdfplumber#460] [Q&A] [answered]
Hi everybody,
I would need some explainations about two properties of the characters.
In particular:

1. I've noticed that the fontnames (e.g. 'BCDEEE+Calibri', 'BCDFEE+Calibri-Bold', 'ADUOFK+TheSansSemiBold-Plain') always have a string followed by "+". I've searched online but nobody knows what that string (e.g. 'ADUOFK') is. Do you have any idea?
2. The second question instead is related to the formula used to get adv.  I don't get what this value actually represent. Isn't the "size" of the char sufficient? 

Thank you very much for your availability!

**Comment by samkit-jain** [comment, marked as answer]
Hi @EmanueleGusso 

1. You may refer to section 9.6.4 (page 266) of the [PDF 1.7 specification](https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/PDF32000_2008.pdf). Those 6 random uppercase letters represent the tag for the font subtype.
2. `pdfplumber` relies on `pdfminer.six` for reading the PDFs and the field `adv` is generated by it and not `pdfplumber` just to clarify. The calculation can be found [here](https://github.com/pdfminer/pdfminer.six/blob/22f90521b823ac5a22785d1439a64c7bdf2c2c6d/pdfminer/layout.py#L291) and represents the scaled width of a character. I am not sure but I think it is mainly used when converting PDF to HTML.
